alonet/detr_panoptic/detr_panoptic.py

```diff
@@ -0,0 +1,222 @@
+# Inspired by the official DETR repository and adapted for aloception
+# https://github.com/facebookresearch/detr/blob/eb9f7e03ed8e2ed2cd55528989fe7df890bc3fc0/models/detr.py
+"""
+Panoptic module to predict object segmentation.
+"""
+import os
+from typing import Callable, Dict
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from alonet.detr.detr_r50 import DetrR50
+from alonet.detr_panoptic.nn import FPNstyleCNN
+from alonet.detr_panoptic.nn import MHAttentionMap
+from alonet.detr_panoptic.utils import get_mask_queries
+from alonet.detr.misc import assert_and_export_onnx
+
+import aloscene
+import alonet
+
+
+class PanopticHead(nn.Module):
+    """Head Pytorch module to predict segmentation masks from previous boxes detection task.
+
+    Parameters
+    ----------
+    DETR_module : alonet.detr.Detr
+        Object detection module based on DETR architecture
+    freeze_detr : bool, optional
+        Freeze DETR_module weights in train procedure, by default True
+    aux_loss: bool, optional
+        Return aux outputs in forward step (if required), by default use DETR_module.aux_loss attribute value
+    device : torch.device, optional
+        Configure module in CPU or GPU, by default torch.device("cpu")
+    weights : str, optional
+        Load weights from name project, by default None
+    """
+
+    INPUT_MEAN_STD = alonet.detr.detr.INPUT_MEAN_STD
+
+    def __init__(
+        self,
+        DETR_module: alonet.detr.Detr,
+        freeze_detr: bool = True,
+        aux_loss: bool = None,
+        device: torch.device = torch.device("cpu"),
+        weights: str = None,
+    ):
+        super().__init__()
+        self.detr = DETR_module
```

Comment:
If the goal is to reuse this panoptic module for deformable as well, then self.detr could be rename by something less specific: `self.base_model` ?

Reply:
@thibo73800 It is correct. But, if I change this parameter, the pre-trained weights won't match (given by the attribute name changed). So, Should I change the pre-trained weights too?

Reply:
I see, let's find a solution fo this issue once we got to adapt the code to handle both detr & Deformable

Reply:
Ok. I will create a new issue to adapt panoptic head to deformable